src/test/java/io/rcktapp/api/handler/sql/SqlGetHandlerTest.java

```diff
@@ -0,0 +1,125 @@
+
+    @Mock
+    SqlDb sqlDb;
+
+    @Mock
+    SqlRql sqlRql;
+
+    @Mock
+    Connection connection;
+
+    @Mock
+    SqlGetHandler handler;
+
+    @Captor
+    ArgumentCaptor<List> queryParametersCaptor;
+
+    @Test
+    public void testSqlGetHandler_queryParametersAreProperlyPassedToQueryObjectsMethod() throws Exception {
+
+        Url url = new Url("http://localhost:8080/api/lift/us/plugins/updatexml(rand(),concat(CHAR(126),user(),CHAR(126)),null)");
+
+        String passedQuery = "updatexml(rand(),concat(CHAR(126),user(),CHAR(126)),null)";
+        String collectionName = "plugins";
+        String [] inClause = {"updatexml(rand()", "concat(CHAR(126)", "user()", "CHAR(126))", "null)"};
+
+        // Setup request object with proper api configuration
+        Request request = new Request(url, "GET", new HashMap<>(), new HashMap<>(), "");
+        Service.ApiMatch apiMatch = new Service.ApiMatch(api, endpoint, "GET", request.getUrl(), "http://localhost:8080/api/lift/us/", collectionName + "/" + passedQuery + "/");
+        request.setApiMatch(apiMatch);
+
+        mockStatic(Rql.class);
+        mockStatic(Sql.class);
+
+        // Mocked methods that return local objects used in the future
+        when(chain.getService()).thenReturn(service);
+        when(service.getDb(request.getApi(), request.getCollectionKey(), SqlDb.class)).thenReturn(sqlDb);
+        when(sqlDb.getConnection()).thenReturn(connection);
+        when(Rql.getRql(sqlDb.getType())).thenReturn(sqlRql);
+        when(api.getCollection(request.getCollectionKey(), SqlDb.class)).thenReturn(collection);
+        when(collection.getEntity()).thenReturn(entity);
+        when(entity.getTable()).thenReturn(table);
+        when(entity.getKey()).thenReturn(attribute);
+        when(attribute.getName()).thenReturn(collectionName);
+        when(Sql.getInClauseStr(J.explode(",", Sql.check(request.getEntityKey())))).thenReturn(passedQuery);
+
+        // When we call the service method on the mocked handler actually call the real method
+        doCallRealMethod().when(handler).service(service, api, endpoint, action, chain, request, response);
+
+        // More mocked methods that return objects reused locally
+        when(sqlRql.asCol(table.getName())).thenReturn("Plugins");
+        when(Sql.getQuestionMarkStr(inClause)).thenReturn("?,?,?,?,?");
+        when(sqlRql.createStmt(eq(" SELECT * FROM Plugins WHERE null IN (?,?,?,?,?) "), any(), eq(request.getParams()), any(Replacer.class))).thenReturn(statement);
+        when(chain.getConfig("maxRows", 0)).thenReturn(0);
+
+        // Call the actual method on the mocked handler
+        handler.service(service,api,endpoint,action,chain,request,response);
+
+        // Verify that the invocation of queryObjects contains the proper parameters.
+        verify(handler).queryObjects(any(), any(), any(), any(), any(), any(), any(), any(), any(), any(), any(), any(), any(), any(), queryParametersCaptor.capture());
```

Comment:
since the solution here is to prevent sql methods from executing by using a parameterized query - perhaps we should start with asserting that no methods were injected in the base query